crates/ibc/src/core/ics24_host/validate.rs

```diff
@@ -48,43 +50,100 @@ pub fn validate_identifier(id: &str, min: usize, max: usize) -> Result<(), Error
     Ok(())
 }
 
+/// Checks if the prefix is valid and can form valid domain identifiers.
+pub fn validate_prefix_identifier(
+    prefix: &str,
+    min_id_length: usize,
+    max_id_length: usize,
+) -> Result<(), Error> {
+    // Checks if the prefix is not blank
+    if prefix.is_empty() {
+        return Err(Error::Empty)?;
+    }
+
+    // Checks of the prefix does not end with a dash
+    let re = safe_regex::regex!(br".*[^-]");
+    if !re.is_match(prefix.as_bytes()) {
+        return Err(Error::InvalidPrefix {
+            prefix: prefix.to_string(),
+        })?;
+    }
+
+    // Checks if the prefix forms a valid default identifier when used with `0`
+    validate_default_identifier(
+        &format!("{prefix}-{}", u64::MIN),
+        min_id_length,
+        max_id_length,
+    )?;
+
+    // Checks if the prefix forms a valid default identifier when used with `u64::MAX`
+    validate_default_identifier(
+        &format!("{prefix}-{}", u64::MAX),
+        min_id_length,
+        max_id_length,
+    )?;
+
+    Ok(())
+}
+
+/// Checks if the identifier is of valid format and can be parsed into the
+/// correct domain identifier type.
+pub fn validate_identifier(id: &str, min: usize, max: usize) -> Result<(), Error> {
+    let split_id: Vec<_> = id.split('-').collect();
+    let last_index = split_id.len() - 1;
+    let prefix = split_id[..last_index].join("-");
+
+    validate_prefix_identifier(prefix.trim(), min, max)?;
+
+    split_id[last_index]
+        .parse::<u64>()
+        .map_err(|_| Error::InvalidCharacter { id: id.into() })?;
+
+    Ok(())
+}
+
+/// Default validator function for the Client types.
+pub fn validate_client_type(id: &str) -> Result<(), Error> {
+    validate_prefix_identifier(id, 9, 64)
+}
+
 /// Default validator function for Client identifiers.
 ///
-/// A valid identifier must be between 9-64 characters and only contain lowercase
-/// alphabetic characters,
+/// A valid client identifier must be between 9-64 characters as specified in
+/// the ICS-24 spec.
 pub fn validate_client_identifier(id: &str) -> Result<(), Error> {
     validate_identifier(id, 9, 64)
```

Comment:
This is no longer correct; client identifiers are not required to be in the `{client_type}-version` format

Reply:
It is the current way that we create client identifiers, which is a subset of all allowed client identifiers. In other words, if ever we changed how we do this in the future (e.g. to allow for other formats), then this `validate_client_identifier()` would fail for valid identifiers. This method should check the validity of client identifiers according to the spec, which will make sure that no matter how we create our client identifiers, they will be valid

Reply:
As per our discussion, needed changes applied. Please check it out again.